# Introduction

Machine learning project for detecting fraudulent credit card transactions using the public dataset from **ULB (Université Libre de Bruxelles)**. This dataset is widely used for research on classification under highly imbalanced conditions.

* Objective

Build a classifier capable of identifying fraudulent transactions, even when they represent less than 0.2% of all records.

# Import Statements

In [19]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

from sklearn.model_selection import train_test_split
from sklearn.preprocessing import StandardScaler
from sklearn.linear_model import LogisticRegression
from sklearn.metrics import classification_report, confusion_matrix


# Data Load

In [20]:
#Load dataset
df = pd.read_csv('../data/raw/creditcard.csv')

# Data Exploration

In [21]:
# Count of fraud and non-fraud transactions
print(df['Class'].value_counts())

Class
0    284315
1       492
Name: count, dtype: int64


#### This dataset has 492 fraud transactions and 284315 non-fraud transactions. Making it an imbalanced problem. Although, the accuracy of the model is relevant to track the frauds.

In [23]:
# Split the dataset into training and testing sets, x being the training set and y being the testing set.
X = df.drop("Class", axis=1)
y = df["Class"]

In [24]:
X_train, X_test, y_train, y_test = train_test_split(
    X, y, test_size=0.2, random_state=42, stratify=y
)
# Splitting the dataset into training and testing sets, with a test size of 20% and 80% training size, acceptable size for a model.